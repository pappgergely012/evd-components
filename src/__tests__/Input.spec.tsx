import React from "react";
import { Input } from "../index";
import renderer, { act } from "react-test-renderer";

test("Input test", () => {
  const component = renderer.create(
    <Input
      placeholder="Name"
      value="George"
      onChange={(val: string) => console.log(val)}
    />
  );
  const instance = component.root;

  expect(instance.findByType(Input).props.placeholder).toBe("Name");
  expect(instance.findByType(Input).props.value).toBe("George");

  act(() =>Â {
    const inst: any = component.toJSON();
    const input: any = inst.children[0].children[1];
    
    input.props.onChange( { target: { value: 'value come from input' } } );
  });

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});