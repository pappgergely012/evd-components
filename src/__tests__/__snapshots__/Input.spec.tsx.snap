// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test 1`] = `
<div
  className="full-input-outer"
>
  <div
    className="input-container not-empty"
  >
    <label
      className="input-placeholder"
    >
      Name
    </label>
    <input
      className="input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="George"
    />
  </div>
</div>
`;
